Revert "Revert unapproved RFC Editor changes" (#27)
rfc-editor/draft-ietf-tls-subcerts: draft-ietf-tls-subcerts.md

---
title: Delegated Credentials for (D)TLS
abbrev:
docname: draft-ietf-tls-subcerts-15
submissionType: IETF
date: January 2023
category: std

ipr: trust200902

> # area: Security
> # workgroup: tls

Reviewer: ```suggestion
```

If these lines are no longer required, just delete them instead of commenting them out.

Author: ```suggestion
area: Security
workgroup: tls
```

Author: We recommend leaving it in because we want this info in the XML file.  We have removed the hash tags so they are longer commented out.

--- abstract

The organizational separation between operators of TLS and DTLS

> endpoints and the certification authority (CA) can create limitations.  For

Reviewer: ```suggestion
endpoints and the certification authority can create limitations.  For
```

This acronym isn't used elsewhere in the Abstract, and it is defined on first use in the body.

long-lived certificates.  If an external Certification Authority (CA) is unable to issue a certificate in
time to replace a deployed certificate, the server would no longer be able to
present a valid certificate to clients.  With short-lived certificates, there is

> a smaller window of time to renew the certificates and therefore a higher risk that

Reviewer: ```suggestion
a smaller window of time to renew a certificate and therefore a higher risk that
```

an outage at a CA will negatively affect the uptime of the TLS-fronted service.

Typically, a (D)TLS server uses a certificate provided by some entity other than
the operator of the server (a CA) {{!RFC8446}}
{{!RFC5280}}.  This organizational separation makes the (D)TLS server operator

> dependent on the CA for some aspects of its operations; for example:

Reviewer: ```suggestion
dependent on the CA for some aspects of its operations. For example:
```

private key.

> Use of short-lived certificates with automated certificate issuance,

Reviewer 2: ```suggestion
The use of short-lived certificates with automated certificate issuance,
```

> Use of short-lived certificates with automated certificate issuance,

Reviewer 2: ```suggestion
The use of short-lived certificates with automated certificate issuance,
```

# Operational Considerations

The operational considerations documented in this section should be

> taken into consideration when using Delegated Certificates.

Reviewer: "Delegated Certificates" here should be "Delegated Credentials".

Reviewer 3: ```suggestion
taken into consideration when using Delegated Credentials.
```

This document registers the "delegated_credential" extension in the
"TLS ExtensionType Values" registry.  The "delegated_credential"

> extension has been assigned code point 34.  The IANA registry

Reviewer: ```suggestion
extension has been assigned the ExtensionType value 34.  The IANA registry
```

lists this extension as "Recommended" (i.e., "Y") and indicates that
it may appear in the ClientHello (CH), CertificateRequest (CR),
or Certificate (CT) messages in (D)TLS 1.3 {{RFC8446}} {{RFC9147}}.
Additionally, the "DTLS-Only" column is assigned the value "N".

This document also defines an ASN.1 module for the DelegationUsage
certificate extension in {{module}}.  IANA has registered value 95 for
"id-mod-delegated-credential-extn" in the "SMI Security for PKIX Module
Identifier" (1.3.5.1.5.5.7.0) registry.  An OID for the DelegationUsage certificate extension
is not needed, as it is already assigned to the extension from
Cloudflare's IANA Private Enterprise Number (PEN) arc.

# Security Considerations

The security considerations documented in this section should be

> taken into consideration when using Delegated Certificates.

Reviewer: {{RFC5280, Section 4.2.1.3}}

Reviewer 3: @bifurcation, please let us know where "{{RFC5280, Section 4.2.1.3}}" should be placed.  Does it replace "this section", or is it additional text?  Also, should "Delegated Certificates" be "Delegated Credentials" here as well?

Reviewer 4: @lbartholomew-rpc this reference does not belong in this section, but I think you are correct Delegated Credentials should replace delegated Certificates in this section and section 5 (Operational Considerations)

Reviewer 2: Delegated Certificates should be Delegated Credentials here. I'm don't think the RFC5280 reference makes sense here as this has been referenced in the appropriate sections. Happy to ignore this comment unless @bifurcation chimes in.

Reviewer: I think this comment was added in error.  Please ignore.

impersonate the compromised party in new TLS connections until the
delegated credential expires.

> Thus, delegated credentials should not be used to send a delegation to an untrusted party; rather they

Reviewer: ```suggestion
Thus, delegated credentials should not be used to send a delegation to an untrusted party. Rather, they
```

## Revocation of Delegated Credentials

Delegated credentials do not provide any additional form of early revocation.

> Since it is short lived, the expiry of the delegated credential revokes

Reviewer: ```suggestion
Since it is short-lived, the expiry of the delegated credential revokes
```

## Privacy Considerations

Delegated credentials can be valid for 7 days (by default), and it is much easier for a
service to create delegated credentials than a certificate signed by a CA.  A
service could determine the client time and clock skew by creating several
delegated credentials with different expiry timestamps and observing whether the

> client would accept it.  Client time could be unique; thus privacy-sensitive
> clients, such as browsers in incognito mode, who do not trust the service might
> not want to advertise support for delegated credentials or limit the number of
> probes that a server can perform.

Reviewer: ```suggestion
client would accept it.  Since client time can be unique to a particular client, privacy-sensitive
clients who do not trust the service, such as browsers in incognito mode, might
not want to advertise support for delegated credentials, or might limit the number of
probes that a server can perform.
```
